src/App.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import { Route, Routes } from "react-router";
+
+import Header from "./components/Header/Header";
+import Home from "./components/Home/Home";
+import SideMenu from "./components/SideMenu/SideMenu";
+import Profile from "./components/Profile/Profile";
+
+import api from "@utilities/api";
+import PostItem from "@components/Home/components/PostFeed/PostItem/PostItem";
+import PulseLoader from "@components/Loader/PulseLoader";
+
+import "./App.css";
+
+function App() {
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    async function getUser() {
+      const fetch = await api.contact.get(1);
+      setUser(await fetch);
+    }
+    getUser();
+  }, []);
+
+  if (!user) {
+    return <PulseLoader />;
+  }
+
+  return (
+    <div className="app-container container mx-auto grid min-h-screen w-screen text-cohort-blue">
+      <Header user={user} />
+      <SideMenu user={user} />
+      <section className="app-main flex flex-col gap-4 bg-cohort-background p-6">
+        <Routes>
+          <Route path="/profile/:contactIdParam" element={<Profile />} />
+          <Route path="/post/:postIdParam" element={<PostItem user={user} />} />
```

Comment:
It's safe to re-use parameter names in separate routes, so these could be:

```js
<Route path="/profile/:id" element={<Profile />} />
<Route path="/post/:id" element={<PostItem user={user} />} />
```

Reply:
I think I did have `path="/post/:id"` at some point. I then changed the name so it was more clear in [`PostItem.jsx` R45-51, for example](https://github.com/boolean-uk/react-cohort-dashboard-challenge/pull/32/files/fa27ac4a141a61b2e3b4aca1727a691fe648ecc3#diff-ff0865388c52b4f64345c2e1503b46c2add9c810765073d8377692a2c1e6d3b3R45-R51) what was being evaluated as there are a few different `ids` floating around in that component.

Would it generally be a given that `id` in a component is coming from a url parameter or should I instead do destructuring like so?: `const { id: postIdParam } = useParams()`

Reply:
Generall with any route parameter you want to keep it as simple as possible. If you're rendering a specific post, it makes sense that the post ID is just called `id` rather than `postId`. There's nothing specifically wrong with using postId as the parameter name, it's just uniform and easy to grasp